htdocs/accountancy/admin/productaccount.php

```diff
@@ -482,11 +528,40 @@
 
 		// Purchases
 		if ($obj->product_type == 0) {
-			$compta_prodbuy = (! empty($conf->global->ACCOUNTING_PRODUCT_BUY_ACCOUNT) ? $conf->global->ACCOUNTING_PRODUCT_BUY_ACCOUNT : $langs->trans("CodeNotDef"));
-			$compta_prodbuy_id = $aarowid_prodbuy;
-		} else {
-			$compta_prodbuy = (! empty($conf->global->ACCOUNTING_SERVICE_BUY_ACCOUNT) ? $conf->global->ACCOUNTING_SERVICE_BUY_ACCOUNT : $langs->trans("CodeNotDef"));
-			$compta_prodbuy_id = $aarowid_servbuy;
+			if ($accounting_product_mode == 'ACCOUNTANCY_BUY') {
+				$compta_prodbuy = (! empty($conf->global->ACCOUNTING_PRODUCT_BUY_ACCOUNT) ? $conf->global->ACCOUNTING_PRODUCT_BUY_ACCOUNT : $langs->trans("CodeNotDef"));
+				$compta_prodbuy_id = $aarowid_prodbuy;
+			}
+			elseif ($accounting_product_mode == 'ACCOUNTANCY_BUY_INTRA') {
+				$compta_prodbuy = (! empty($conf->global->ACCOUNTING_PRODUCT_BUY_INTRA_ACCOUNT) ? $conf->global->ACCOUNTING_PRODUCT_BUY_INTRA_ACCOUNT : $langs->trans("CodeNotDef"));
+				$compta_prodbuy_id = $aarowid_prodbuy_intra;
+			}
+			elseif ($accounting_product_mode == 'ACCOUNTANCY_BUY_EXPORT') {
+				$compta_prodbuy = (! empty($conf->global->ACCOUNTING_PRODUCT_BUY_EXPORT_ACCOUNT) ? $conf->global->ACCOUNTING_PRODUCT_BUY_EXPORT_ACCOUNT : $langs->trans("CodeNotDef"));
+				$compta_prodbuy_id = $aarowid_prodbuy_export;
+			}
+			else {
+				$compta_prodbuy = (! empty($conf->global->ACCOUNTING_SERVICE_BUY_ACCOUNT) ? $conf->global->ACCOUNTING_SERVICE_BUY_ACCOUNT : $langs->trans("CodeNotDef"));
```

Comment:
product and services are mixed here.
We should have ACCOUNTING_PRODUCT_BUY_ACCOUNT here ?

Reply:
Yes, like the system for sales